sql/api/src/main/scala/org/apache/spark/sql/api/TableValuedFunction.scala

```diff
@@ -0,0 +1,177 @@
+   * @since 4.0.0
+   */
+  def inline_outer(input: Column): Dataset[Row]
+
+  /**
+   * Creates a `DataFrame` containing a new row for a json column according to the given field
+   * names.
+   *
+   * @group json_funcs
+   * @since 4.0.0
+   */
+  @scala.annotation.varargs
+  def json_tuple(input: Column, fields: Column*): Dataset[Row]
+
+  /**
+   * Creates a `DataFrame` containing a new row for each element with position in the given array
+   * or map column. Uses the default column name `pos` for position, and `col` for elements in the
+   * array and `key` and `value` for elements in the map unless specified otherwise.
+   *
+   * @group generator_funcs
+   * @since 4.0.0
+   */
+  def posexplode(collection: Column): Dataset[Row]
+
+  /**
+   * Creates a `DataFrame` containing a new row for each element with position in the given array
+   * or map column. Uses the default column name `pos` for position, and `col` for elements in the
+   * array and `key` and `value` for elements in the map unless specified otherwise. Unlike
+   * posexplode, if the array/map is null or empty then the row (null, null) is produced.
+   *
+   * @group generator_funcs
+   * @since 4.0.0
+   */
+  def posexplode_outer(collection: Column): Dataset[Row]
+
+  /**
+   * Separates `col1`, ..., `colk` into `n` rows. Uses column names col0, col1, etc. by default
+   * unless specified otherwise.
+   *
+   * @group generator_funcs
+   * @since 4.0.0
+   */
+  @scala.annotation.varargs
+  def stack(n: Column, fields: Column*): Dataset[Row]
+
+  /**
+   * Gets all of the Spark SQL string collations.
+   *
+   * @group generator_funcs
+   * @since 4.0.0
+   */
+  def collations(): Dataset[Row]
+
+  /**
+   * Gets Spark SQL keywords.
+   *
+   * @group generator_funcs
+   * @since 4.0.0
+   */
+  def sql_keywords(): Dataset[Row]
+
+  /**
+   * Separates a variant object/array into multiple rows containing its fields/elements. Its
+   * result schema is `struct&lt;pos int, key string, value variant&gt;`. `pos` is the position of
+   * the field/element in its parent object/array, and `value` is the field/element value. `key`
+   * is the field name when exploding a variant object, or is NULL when exploding a variant array.
+   * It ignores any input that is not a variant array/object, including SQL NULL, variant null,
+   * and any other variant values.
+   *
+   * @group variant_funcs
+   * @since 4.0.0
+   */
+  def variant_explode(input: Column): Dataset[Row]
+
+  /**
+   * Separates a variant object/array into multiple rows containing its fields/elements. Its
+   * result schema is `struct&lt;pos int, key string, value variant&gt;`. `pos` is the position of
+   * the field/element in its parent object/array, and `value` is the field/element value. `key`
+   * is the field name when exploding a variant object, or is NULL when exploding a variant array.
+   * It ignores any input that is not a variant array/object, including SQL NULL, variant null,
```

Comment:
It doesn't ignore those inputs, but produces NULLs instead, right?

Reply:
Thanks!